### PR TITLE
Crazy tar code

### PR DESCRIPTION
**Pull request by nturaga**
Compare with original and only merge when fixed

### CODE REVIEW

File: aRchive.py [hunk 1]
```diff
@@ -143,13 +151,27 @@ def archive_package_versions(bioc_pack, archive_dir):
             log.debug("Bioc_pack version %s %s" % (bioc_pack, curr_version))
             # Create new directory with version number as "-version" extension
             bioc_pack_name = os.path.split(bioc_pack)[-1]
-            output_directory = os.path.join(archive_dir,
-                                            bioc_pack_name, curr_version)
-            if not os.path.exists(output_directory):
-                log.info("Made new version directory: %s" % output_directory)
+            output_directory = os.path.join(archive_dir)
+            out_tarfile = bioc_pack_name + "_" + curr_version + '.tar.gz'
```
Comment (hexylena) [inline aRchive.py]: might switch to formatting a string `"%s_%s.tar.gz" % (bioc_pack_name, curr_version)`

File: aRchive.py [hunk 2]
```diff
@@ -143,13 +151,27 @@ def archive_package_versions(bioc_pack, archive_dir):
...
+
+            if not os.path.exists(dest_tar_file):
                 # SAVE THE CURRENT VERSION HERE
-                shutil.copytree(bioc_pack, output_directory)
+                # Tar the directory
+                log.info('adding contents of bioc_pack %s to tarfile %s' % (bioc_pack, out_tarfile))
+                make_tarfile(out_tarfile, bioc_pack)
+                dest = os.path.join(output_directory, out_tarfile)
+                log.info('Moving %s to %s' % (out_tarfile, dest))
+                shutil.move(out_tarfile, dest)
```
Comment (hexylena) [inline aRchive.py:175]: Any reason to move the tar? Why not create it in the destination location in the first place.

File: aRchive.py [hunk 3]
```diff
@@ -24,8 +24,9 @@
 import subprocess
 import shutil
 import re
+import tarfile
 import logging
-logging.basicConfig(level=logging.DEBUG, name="aRchive")
+logging.basicConfig(level=logging.DEBUG, name="aRchive", filename="archive.log")
```
Comment (hexylena) [inline aRchive.py:29]: +1, the logs for this tool are quite noisy.

**Comment by hexylena** [comment]
will test shortly.

File: aRchive.py (hunk 2 above)
Comment (nturaga) [inline aRchive.py:175]: Just to prevent the Bioc base folder and the archive from mixing. No concrete reason otherwise.

**Comment by hexylena** [comment]
nice!

**Pull request merged**